src/node/rpc/node_frontend.h

```diff
@@ -1275,6 +1275,9 @@ namespace ccf
           in.certificate_signing_request,
           in.public_key};
 
+        auto tx_ = static_cast<kv::CommittableTx*>(&ctx.tx);
+        tx_->set_flag(kv::AbstractStore::Flag::LEDGER_CHUNK_BEFORE_THIS_TX);
```

Comment:
Why do we need to cast?

Reply:
We don't want to expose the `set_flag()` API to the `kv::Tx` used by application endpoints, hence the cast to the internal `CommittableTx` type here.

Reply:
We should check that `tx_` is not nullptr after the static cast.